client/components/phone-input/country-flag.jsx

```diff
@@ -5,68 +5,60 @@
  */
 
 import PropTypes from 'prop-types';
-import React from 'react';
+import React, { Component } from 'react';
 import Gridicon from 'gridicons';
-
 /** Internal Dependencies */
 import Spinner from 'components/spinner';
 
-export default class extends React.Component {
+export default class extends Component {
 	static displayName = 'PhoneInputCountryFlag';
 
 	static propTypes = {
 		countryCode: PropTypes.string.isRequired,
 	};
 
 	state = {
-		ready: false,
-		error: false,
+		flagSvg: null,
 	};
 
-	componentDidUpdate( oldProps ) {
-		if ( this.props.countryCode && this.props.countryCode !== oldProps.countryCode ) {
-			this.setState( { ready: false, error: false } );
-		}
+	componentDidMount() {
+		this.updateFlag( null, this.props.countryCode );
 	}
 
-	renderSpinner = () => {
-		if ( ( ! this.props.countryCode || ! this.state.ready ) && ! this.state.error ) {
-			return <Spinner size={ 16 } className="phone-input__flag-spinner" />;
-		}
-	};
-
-	handleImageLoad = () => {
-		this.setState( { ready: true, error: false } );
-	};
-
-	handleImageError = () => {
-		this.setState( { ready: false, error: true } );
-	};
+	componentDidUpdate( prevProps ) {
+		this.updateFlag( prevProps.countryCode, this.props.countryCode );
+	}
 
-	renderFlag = () => {
-		const style = this.state.ready ? {} : { visibility: 'hidden' };
+	updateFlag = ( prevCode, nextCode ) => {
+		if ( prevCode !== nextCode ) {
+			// don't confusingly split this import statement with its long string
+			// prettier-ignore
+			import(
+				/* webpackChunkName: "svg-flag-[request]" */
+				'svg-url-loader?noquotes!../../../node_modules/flag-icon-css/flags/4x3/' + nextCode + '.svg'
+			).then(
+				( { default: flagSvg } ) => this.setState( { flagSvg } ),
+				() => this.setState( { flagSvg: false } ),
+			);
 
-		if ( this.props.countryCode ) {
-			if ( ! this.state.error ) {
-				return (
-					<img
-						onLoad={ this.handleImageLoad }
-						onError={ this.handleImageError }
-						src={ `/calypso/images/flags/${ this.props.countryCode }.svg` }
-						className="phone-input__flag-icon"
-						style={ style }
-					/>
-				);
-			}
-			return <Gridicon icon="globe" size={ 24 } className="phone-input__flag-icon" />;
+			// called synchronously before the dynamic import loads
+			this.setState( { flagSvg: null } );
```

Comment:
If it's called before, can we move it before the `import()`? Possibly into `componentDidUpdate()` (since `componentDidMount()` will use the initial `state` which is `null` anyway).

Reply:
done